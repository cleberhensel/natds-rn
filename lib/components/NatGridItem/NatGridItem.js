import React from 'react';
import PropTypes from 'prop-types';
import { TouchableWithoutFeedback } from 'react-native';

import {
  Thumb, Item, Description, Title,
} from './NatGridItem.styles';

export const NatGridItem = ({
  source, sizeMedium, onPress, children,
}) => (
  <Item sizeMedium={sizeMedium}>
    <TouchableWithoutFeedback onPress={onPress}>
      <Thumb
        accessibilityRole="image"
        sizeMedium={sizeMedium}
        source={{ uri: source }}
        accessibilityLabel={`${children}`}
      />
    </TouchableWithoutFeedback>
    <Description
      sizeMedium={sizeMedium}
      accessibilityLabel={`${children}`}
    >
      <Title>{children}</Title>
    </Description>
  </Item>
);

NatGridItem.propTypes = {
  children: PropTypes.string,
  onPress: PropTypes.func,
  sizeMedium: PropTypes.bool,
  source: PropTypes.string.isRequired,
};

NatGridItem.defaultProps = {
  children: 'TÃ­tulo da imagem',
  onPress: PropTypes.func,
  sizeMedium: false,
};
