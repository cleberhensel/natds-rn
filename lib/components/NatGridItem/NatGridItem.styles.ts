import { Platform } from 'react-native';
import styled from 'styled-components/native';
import { roboto, sanfrancisco } from '../../assets/theme/fonts/fonts';
import { colors } from '../../assets/theme/colors/colors';

export const Item = styled.View`
  margin-bottom: ${props => (props.sizeMedium ? 24 : 0)};
  margin-right: ${props => (props.sizeMedium ? 12 : 16)};
  margin-left: ${props => (props.sizeMedium ? 12 : 0)};
  width: ${props => (props.sizeMedium ? 156 : 128)};
  height: ${props => (props.sizeMedium ? 200 : 176)};
`;
export const Thumb = styled.Image`
  width: ${props => (props.sizeMedium ? 156 : 128)};
  height: ${props => (props.sizeMedium ? 156 : 128)};
  border-radius: 6;
  margin-bottom: 8px;
`;
export const Description = styled.View`
  width: ${props => (props.sizeMedium ? 156 : 128)};
  height: 40px;
  line-height: 20;
`;
export const Title = styled.Text`
  flex: 1;
  font-size: 14;
  font-weight: 500;
  font-style: normal;
  text-align: center;
  color: ${colors.bw.grayDark};
  font-family: ${Platform.OS === 'ios' ? sanfrancisco.regular : roboto.regular};
`;
