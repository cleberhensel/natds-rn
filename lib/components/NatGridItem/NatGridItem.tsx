import React from 'react';
import PropTypes from 'prop-types';
import { TouchableWithoutFeedback } from 'react-native';
import {
  Thumb, Item, Description, Title,
} from './NatGridItem.styles';

type NatGridItemProps = {
  onPress?: (...args: any[]) => any,
  sizeMedium?: boolean,
  source: string
};

export const NatGridItem: React.SFC<NatGridItemProps> = ({
  source,
  sizeMedium,
  onPress,
  children,
}) => (
  <Item sizeMedium={sizeMedium}>
    <TouchableWithoutFeedback onPress={onPress}>
      <Thumb
        accessibilityRole="image"
        sizeMedium={sizeMedium}
        source={{ uri: source }}
        accessibilityLabel={`${children}`}
      />
    </TouchableWithoutFeedback>
    <Description sizeMedium={sizeMedium} accessibilityLabel={`${children}`}>
      <Title>{children}</Title>
    </Description>
  </Item>
);
NatGridItem.defaultProps = {
  children: 'TÃ­tulo da imagem',
  onPress: PropTypes.func,
  sizeMedium: false,
};
