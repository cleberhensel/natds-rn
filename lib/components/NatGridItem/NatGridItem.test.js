import React from 'react';
import renderer from 'react-test-renderer';

import { NatGridItem } from './NatGridItem';
import {
  Thumb, Item, Description, Title,
} from './NatGridItem.styles';
import 'jest-styled-components/native';

describe('NatGridItem and local styled components', () => {
  describe('NatGridItem component', () => {
    it('Should render NatGridItem component ', () => {
      const griditem = renderer.create(
        <NatGridItem sizeMedium onPress={jest.fn()} source="#">
          Texto
        </NatGridItem>,
      ).toJSON();

      expect(griditem).toMatchSnapshot();
    });

    describe('Properties of NatGridItem', () => {
      let press = () => {};

      beforeEach(() => {
        press = jest.fn();
      });

      const props = {
        children: 'descrição',
        onPress: press,
        sizeMedium: true,
        source: '#',
      };

      it('Should render NatGridItem component ', () => {
        const griditem = renderer.create(<NatGridItem {...props} />).toJSON();

        expect(griditem).toMatchSnapshot();
      });
    });
  });

  describe('Style Thumb', () => {
    it('Thumb default', () => {
      const thumb = renderer.create(<Thumb />).toJSON();

      expect(thumb).toHaveStyleRule('width', 128);
      expect(thumb).toHaveStyleRule('height', 128);
      expect(thumb).toHaveStyleRule('border-radius', 6);
      expect(thumb).toHaveStyleRule('margin-bottom', 8);
    });

    it('Style Thumb with sizeMedium props', () => {
      const thumb = renderer.create(<Thumb sizeMedium />).toJSON();

      expect(thumb).toHaveStyleRule('width', 156);
      expect(thumb).toHaveStyleRule('height', 156);
    });
  });

  describe('Style Item', () => {
    it('Item default', () => {
      const item = renderer.create(<Item />).toJSON();

      expect(item).toHaveStyleRule('margin-right', 16);
      expect(item).toHaveStyleRule('width', 128);
      expect(item).toHaveStyleRule('height', 176);
    });

    it('Item with sizeMedium props', () => {
      const item = renderer.create(<Item sizeMedium />).toJSON();

      expect(item).toHaveStyleRule('width', 156);
      expect(item).toHaveStyleRule('height', 200);
    });
  });

  describe('Style Description', () => {
    it('Description default', () => {
      const description = renderer.create(<Description />).toJSON();

      expect(description).toHaveStyleRule('width', 128);
    });

    it('description with sizeMedium props', () => {
      const description = renderer.create(<Description sizeMedium />).toJSON();

      expect(description).toHaveStyleRule('width', 156);
    });
  });

  describe('Style Title', () => {
    it('Title default', () => {
      const title = renderer.create(<Title />).toJSON();

      expect(title).toHaveStyleRule('flex');
      expect(title).toHaveStyleRule('font-size', 14);
      expect(title).toHaveStyleRule('font-weight', '500');
      expect(title).toHaveStyleRule('font-style', 'normal');
      expect(title).toHaveStyleRule('font-align');
      expect(title).toHaveStyleRule('color', '#777777');
      expect(title).toHaveStyleRule('robotoRegular');
    });
  });
});
