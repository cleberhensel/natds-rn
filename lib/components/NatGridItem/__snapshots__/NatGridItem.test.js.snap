// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NatGridItem and local styled components NatGridItem component Properties of NatGridItem Should render NatGridItem component  1`] = `
<View
  sizeMedium={true}
  style={
    Array [
      Object {
        "height": 200,
        "marginBottom": 24,
        "marginLeft": 12,
        "marginRight": 12,
        "width": 156,
      },
    ]
  }
>
  <Image
    accessibilityLabel="descrição"
    accessibilityRole="image"
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    sizeMedium={true}
    source={
      Object {
        "uri": "#",
      }
    }
    style={
      Array [
        Object {
          "borderRadius": 6,
          "height": 156,
          "marginBottom": 8,
          "width": 156,
        },
      ]
    }
  />
  <View
    accessibilityLabel="descrição"
    sizeMedium={true}
    style={
      Array [
        Object {
          "height": 40,
          "lineHeight": 20,
          "width": 156,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#777777",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "fontFamily": "System",
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "500",
            "textAlign": "center",
          },
        ]
      }
    >
      descrição
    </Text>
  </View>
</View>
`;

exports[`NatGridItem and local styled components NatGridItem component Should render NatGridItem component  1`] = `
<View
  sizeMedium={true}
  style={
    Array [
      Object {
        "height": 200,
        "marginBottom": 24,
        "marginLeft": 12,
        "marginRight": 12,
        "width": 156,
      },
    ]
  }
>
  <Image
    accessibilityLabel="Texto"
    accessibilityRole="image"
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    sizeMedium={true}
    source={
      Object {
        "uri": "#",
      }
    }
    style={
      Array [
        Object {
          "borderRadius": 6,
          "height": 156,
          "marginBottom": 8,
          "width": 156,
        },
      ]
    }
  />
  <View
    accessibilityLabel="Texto"
    sizeMedium={true}
    style={
      Array [
        Object {
          "height": 40,
          "lineHeight": 20,
          "width": 156,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#777777",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "fontFamily": "System",
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "500",
            "textAlign": "center",
          },
        ]
      }
    >
      Texto
    </Text>
  </View>
</View>
`;
